feat: ordered playlist results

daibhin opened the pull request:
## Problem

Right now we only ever order recordings by "latest" but is that always the most interesting 🤔 Think about these use cases:

- When looking at sessions containing console errors
  - currently: we return the latest session with any errors
  - proposed: we allow you to order so sessions with the most errors are shown first
- When watching sessions of users who went through the onboarding flow
  - currently: we show all sessions, starting with the latest
  - proposed: we should let you order by duration to show users who took the longest and hence maybe struggled with the flow

## Changes

https://github.com/PostHog/posthog/assets/6685876/12360cde-635a-4d1c-8f55-88ba0982c057

- Feature flagged
- Rely on `target_entity_order` clause

## How did you test this code?

Front and backend tests

Commit:
feat: ordered playlist results

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Commit:
Merge branch 'master' into dn-feat/ordered-playlist-results

Commit:
Merge branch 'master' into dn-feat/ordered-playlist-results

Commit:
add panel in UI

Commit:
only show ordering dropdown when ordering by latest

Commit:
persist ordering per playlist

Commit:
order list via sort

Commit:
cleanup

Commit:
add tests

Commit:
add frontend test

Commit:
better test name

Commit:
update snapshots

Commit:
Merge branch 'master' into dn-feat/ordered-playlist-results

Commit:
fix frontent test mock to allow for offset of 0

tomasfarias committed:
fix: Move join table creation to different client (#20606)

* fix: Move join table creation to different client

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tomasfarias committed:
chore: Add NoSuchBucket to non-retryable (#20607)

mariusandra committed:
feat(trends): aggregate charts and formulas (#20594)

tomasfarias committed:
fix: Create table in cluster (#20608)

Commit:
fix backend test

Commit:
add ordering test

Commit:
add feature flag

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

benjackwhite committed:
revert: "feat(trends): aggregate charts and formulas (#20594)" (#20619)

* Revert "feat(trends): aggregate charts and formulas (#20594)"

This reverts commit 99bd37300c3cb0702ed7e7a93dd50cb213c1a959.

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tiina303 committed:
revert: "feat: allow linked flag variants" (#20623)

* Revert "feat: allow linked flag variants (#20543)"

This reverts commit d1a534ec8332d027d992de35f8b556b483911780.

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

tiina303 committed:
fix: Include rusty hook app metrics (#20605)

bretthoerner committed:
chore(batch-exports): add temporal settings for some clickhouse options (#20622)

pauldambra committed:
chore: even more logging (#20612)

* chore: even more embeddings logging

* and more settings

* fix

* fix

daibhin committed:
chore: count tokens before hitting OpenAI (#20621)

* chore: count tokens before hitting OpenAI

* log the offending input

---------

Co-authored-by: Paul D'Ambra <paul@posthog.com>

tiina303 committed:
revert: "revert: "feat(trends): aggregate charts and formulas (#20594)"" (#20625)

* Revert "revert: "feat(trends): aggregate charts and formulas (#20594)" (#20619)"

This reverts commit 88e68650ab7e23078cd0e023ba9d087c0195c48e.

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

xvello committed:
feat(ingestion): pass PLUGIN_SERVER_MODE as pg app name (#20613)

mariusandra committed:
fix(insights): show more query errors (#20611)

webjunkie committed:
fix(exports): Fix xlsx export extension (#20629)

pauldambra committed:
fix: query needs a max time range (#20626)

tomasfarias committed:
fix: Reload dictionary on cluster (#20630)

liyiy committed:
fix(surveys): targeting flag not deleted during updates bug (#20427)


Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

pauldambra commented on frontend/src/lib/constants.tsx:
```diff
@@ -204,6 +204,7 @@ export const FEATURE_FLAGS = {
     SAVED_NOT_PINNED: 'saved-not-pinned', // owner: #team-replay
     EXPORTS_SIDEPANEL: 'exports-sidepanel', // owner: #team-product-analytics
     SESSION_REPLAY_V3_INGESTION_PLAYBACK: 'session-replay-v3-ingestion-playback', // owner: @benjackwhite
+    SESSION_REPLAY_FILTER_ORDERING: 'session-replay-filter-ordering', // owner: @benjackwhite
```
```suggestion
    SESSION_REPLAY_FILTER_ORDERING: 'session-replay-filter-ordering', // owner: @team-replay
```

pauldambra commented on frontend/src/scenes/session-recordings/filters/OrderingFilters.tsx:
```diff
@@ -0,0 +1,34 @@
+            options={[
+                {
+                    value: 'start_time',
+                    label: 'Latest',
```
should we have data-attrs here so that autocapture can be turned into nice actions to track usage?

pauldambra reviewed (approved):
LGTM

webjunkie committed:
fix(insights): Increase max AST size for funnel time to convert (#20628)

daibhin committed:
Update frontend/src/lib/constants.tsx

Co-authored-by: Paul D'Ambra <paul@posthog.com>

pauldambra committed:
chore: update snapshot fangling (#20634)

* chore: update snapshot fangling

* Update query snapshots

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

mariusandra committed:
fix(frontend): no more "nope" (#20610)

tiina303 committed:
chore: skip falky autocapture functional test (#20638)

chore: skip falky autocapture test

We're not really touching that code atm, this test flakes quite a bit, let's skip it.

pauldambra committed:
fix: clarify masking function (#20635)

Commit:
fix: make advanced filters more obvious (#20631)

Commit:
Merge branch 'master' into dn-feat/ordered-playlist-results

Commit:
add data-attr

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
